### PR TITLE
feat(website): auto-generate rule docs pages

### PR DESCRIPTION
**Pull request by DonIsaac**
> AI-generated description because I'm lazy
### TL;DR

This PR introduces the ability to generate documentation for linter rules and adds new methods and metadata for rule fix capabilities.

To see what this looks like, please check out https://github.com/oxc-project/oxc-project.github.io/pull/165.

## Screenshots
Hyperlinks to rule doc pages in auto-generated rules table
image

Example of a docs page
image

### What changed?

- Added `RuleFixMeta` to indicate rule fix capabilities
- Introduced methods `is_none` and `is_pending` in `RuleFixMeta`
- Modified `render_markdown_table` in `RuleTableSection` to accept an optional link prefix
- Created new modules for rule documentation and HTML rendering
- Updated `print_rules` function to generate markdown for rules and detailed documentation pages

### How to test?

Run the `linter-rules` task with appropriate arguments to generate the markdown table and documentation pages.
Verify the generated files for correctness and that all metadata is correctly displayed.

### Why make this change?

To enhance the project documentation and provide clear rule fix capabilities, thereby improving the developer experience and easing the integration process.

---

### CODE REVIEW

**Comment by DonIsaac** [comment]
* **#4640** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4640?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tasks/website/src/linter/rules/doc_page.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        page,
+        "<!-- This file is auto-generated by {}. Do not edit it manually. -->\n",
+        file!()
+    )?;
+    writeln!(page, "# {plugin}/{name}\n")?;
+
+    // rule metadata
+    page.div(r#"class="rule-meta""#, |p| {
+        if *turned_on_by_default {
+            p.span(r#"class="default-on""#, |p| {
```
Comment (DonIsaac) [inline tasks/website/src/linter/rules/doc_page.rs:27]: We should add an Alert component to the website and use that.

File: tasks/website/src/linter/rules/mod.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Arguments:
+    -t,--table <path>     Path to file where rule markdown table will be saved.
+    -r,--rule-docs <path> Path to directory where rule doc pages will be saved.
+                          A directory will be created if one doesn't exist.
+    -h,--help             Show this help message.
+
+";
+
+/// `cargo run -p website linter-rules > /path/to/oxc/oxc-project.github.io/src/docs/guide/usage/linter/generated-rules.md`
```
Comment (DonIsaac) [inline tasks/website/src/linter/rules/mod.rs:27]: TODO: update

**Comment by DonIsaac** [comment]
I think #1783 will help us add documentation for rule configs. We can implement `JsonSchema` for those configs, then add `config_schema(&self) -> Option<Self::Schema>` to `RuleMeta` to render config sections.

**Comment by DonIsaac** [comment]
I think we also need tests on rule documentation to ensure they produce valid markdown. I'm seeing errors in the website's build pipeline.

**Comment by Boshen** [comment]
Let me defer this merge to the next release. I'll work on releasing the current version first.

**Pull request merged**